src/main/java/com/mineboundteam/minebound/event/ModEvents.java

```diff
@@ -0,0 +1,126 @@
+    @SubscribeEvent
+    public static void onAttachCapabilitiesPlayer(AttachCapabilitiesEvent<Entity> event){
+        if(event.getObject() instanceof Player){
+            if(!event.getObject().getCapability(PlayerManaProvider.PLAYER_MANA).isPresent()){
+                event.addCapability(new ResourceLocation(MineBound.MOD_ID, "properties"), new PlayerManaProvider());
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public static void onPlayerCloned(PlayerEvent.Clone event){
+        if(event.isWasDeath()){
+            event.getOriginal().getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(oldStore ->
+                event.getOriginal().getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(newStore ->
+                    newStore.copyFrom(oldStore)
+                )
+            );
+        }
+    }
+
+    @SubscribeEvent
+    public static void onRegisterCapabilities(RegisterCapabilitiesEvent event){
+        event.register(PlayerMana.class);
+    }
+
+    @SubscribeEvent
+    public static void onPlayerTick(TickEvent.PlayerTickEvent event){
+        if(event.side == LogicalSide.SERVER && event.player.level.getDayTime() % 10 == 0 && event.phase == TickEvent.Phase.START) {
+            event.player.getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(mana -> {
+                int manaBoost = 0;
+                int recBoost = 0;
+                Random rndm = new Random();
+
+                boolean armorSet = true;
+                ArmorTier tier = null;
+                List<ItemStack> mArmors = new ArrayList<>();
+
+                // check each armor slot and get their bonuses to recovery and cap
+                for (EquipmentSlot slot : EnumSet.of(EquipmentSlot.HEAD, EquipmentSlot.CHEST, EquipmentSlot.LEGS, EquipmentSlot.FEET)) {
+                    var item = event.player.getItemBySlot(slot).getItem();
+
+                    if (item instanceof MyrialArmorItem) {
+                        mArmors.add(event.player.getItemBySlot(slot));
+                        var config = ((MyrialArmorItem) item).getConfig();
+                        manaBoost += config.MANAPOOL.get();
+                        recBoost += config.RECOVERY.get();
+
+                        if (tier == null)
+                            tier = ((MyrialArmorItem) item).getTier();
+                        else if (tier != ((MyrialArmorItem) item).getTier())
+                            armorSet = false;
+                        continue;
+                    }
+                    armorSet = false;
+                }
+
+                if (armorSet) {
+                    var setConfig = ArmorConfigRegistry.SET_BONUS_MAP.get(tier);
+                    manaBoost += setConfig.MANAPOOL.get();
+                    recBoost += setConfig.RECOVERY.get();
+                }
+
+                // if mana is recovered, calculate charge drained from armor durability
+                if(mana.getManaMax() + manaBoost > mana.getMana() && rndm.nextInt(10) == 0){
```

Comment:
`if(mana.getManaMax() + manaBoost > mana.getMana()`
Wouldn't this part always be true? Since the mantis bounded by its max? Or is this supposed to be if there is no mana calculate durability loss

Reply:
this is true in every single case EXCEPT when mana is full (i.e. if the current mana is equal to, instead of less than, the cap)